From ef7a81977d76ff520c9d2c6d25a6a0383a988fde Mon Sep 17 00:00:00 2001
From: Seungha Yang <seungha.yang@navercorp.com>
Date: Fri, 8 Feb 2019 21:38:04 +0900
Subject: [PATCH 26/45] glupload: Don't leak caps features

Create caps features when it is required.

(cherry picked from commit 3152cbb46e34950cce55007651ca62c967607686)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst-libs/gst/gl/gstglupload.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 70259ce66f..8a63a2cbb2 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -520,9 +520,7 @@ _dma_buf_upload_transform_caps (gpointer impl, GstGLContext * context,
     GstPadDirection direction, GstCaps * caps)
 {
   struct DmabufUpload *dmabuf = impl;
-  GstCapsFeatures *passthrough =
-      gst_caps_features_from_string
-      (GST_CAPS_FEATURE_META_GST_VIDEO_OVERLAY_COMPOSITION);
+  GstCapsFeatures *passthrough;
   GstCaps *ret;
 
   if (context) {
@@ -534,6 +532,9 @@ _dma_buf_upload_transform_caps (gpointer impl, GstGLContext * context,
       return NULL;
   }
 
+  passthrough = gst_caps_features_from_string
+      (GST_CAPS_FEATURE_META_GST_VIDEO_OVERLAY_COMPOSITION);
+
   if (direction == GST_PAD_SINK) {
     GstCaps *tmp;
 
-- 
2.11.0

